public/scripts/extensions/connection-manager/index.js

```diff
@@ -472,29 +508,57 @@ async function renderDetailsContent(detailsContent) {
         await eventSource.emit(event_types.CONNECTION_PROFILE_LOADED, NONE);
     });
 
-    const renameButton = document.getElementById('rename_connection_profile');
-    renameButton.addEventListener('click', async () => {
+    const editButton = document.getElementById('edit_connection_profile');
+    editButton.addEventListener('click', async () => {
         const selectedProfile = extension_settings.connectionManager.selectedProfile;
         const profile = extension_settings.connectionManager.profiles.find(p => p.id === selectedProfile);
         if (!profile) {
             console.log('No profile selected');
             return;
         }
 
-        const newName = await Popup.show.input('Enter a new name', null, profile.name, { rows: 2 });
+        if (!Array.isArray(profile.exclude)) {
+            profile.exclude = [];
+        }
+
+        const commands = profile.mode === 'cc' ? CC_COMMANDS : TC_COMMANDS;
+        const settings = commands.reduce((acc, command) => {
+            const fancyName = FANCY_NAMES[command];
+            acc[fancyName] = !profile.exclude.includes(command);
+            return acc;
+        }, {});
+        const template = $(await renderExtensionTemplateAsync(MODULE_NAME, 'edit', { name: profile.name, settings }));
+        const newName = await callGenericPopup(template, POPUP_TYPE.INPUT, profile.name);
+
         if (!newName) {
             return;
         }
 
-        if (extension_settings.connectionManager.profiles.some(p => p.name === newName)) {
+        if (profile.name !== newName && extension_settings.connectionManager.profiles.some(p => p.name === newName)) {
             toastr.error('A profile with the same name already exists.');
             return;
         }
 
-        profile.name = newName;
+        const newExcludeList = template.find('input[name="exclude"]:not(:checked)').map(function () {
+            return Object.entries(FANCY_NAMES).find(x => x[1] === String($(this).val()))?.[0];
+        }).get();
+
+        if (newExcludeList.length !== profile.exclude.length || !newExcludeList.every(e => profile.exclude.includes(e))) {
+            profile.exclude = newExcludeList;
+            for (const command of newExcludeList) {
+                delete profile[command];
+            }
+            toastr.info('Press "Update" to record them into the profile.', 'Included settings list updated');
```

Comment:
Why does it say this here? Doesn't the profile get saved below anyway?

Reply:
Wait. That means it only saves the toggles for the profile, but doesn't update stuff like sampler preset or all others if they have changed?
I think I get it.

Instead of having another confirmation, why not add a checkbox at the bottom of the toggle popup where you can check "Update Values" or something?